lib/progressive_hash_diff.rb

```diff
@@ -0,0 +1,30 @@
+class ProgressiveHashDiff
```

Comment:
As I suggested this name, it's probably appropriate I reply.

The purpose of the class is to build up the components of a diff gradually, I.e. in a progressive manner. So at the start of a block of code you can set the first item and later can set the other one (which could be same object in a different state - which is what we do). So this class is almost used to gradually build up what will be diffed, as opposed to us needing to store variables of the components.

Did that explanation make sense?